ntpd/src/daemon/spawn/mod.rs

```diff
@@ -272,24 +259,43 @@ where
         action_tx: mpsc::Sender<SpawnEvent>,
         mut system_notify: mpsc::Receiver<SystemEvent>,
     ) -> Result<(), E> {
-        use tokio::time::{timeout, Duration};
-        // basic event loop where init is called on startup and then wait for
-        // events from the system before doing anything
-        self.handle_init(&action_tx).await?;
-        while let Some(event) = timeout(Duration::from_secs(60), system_notify.recv())
-            .await
-            .unwrap_or(Some(SystemEvent::Idle))
-        {
+        let mut has_ticket = true;
+        let mut last_ticket_time = Instant::now();
+
+        loop {
+            if Instant::now() - last_ticket_time >= NETWORK_WAIT_PERIOD {
+                has_ticket = true;
+            }
+
+            if has_ticket && !self.is_complete() {
+                self.try_spawn(&action_tx).await?;
```

Comment:
Do we still want to add any sort of timeout mechanism here? What are the max timeouts for DNS and for NTS handshake? Is it ok if we wait that long and not receive any system_notify events? I think we should at least add a little more documentation for try_spawn to remind implementers that they should try and keep the spawning logic to as short a time a possible.